### PR TITLE
Upgrade to Workarea v3.5 & Friends

### PR DESCRIPTION
**Pull request by meowsus**
ONETHEME-1

### CODE REVIEW

File: workarea-one_theme.gemspec [hunk 1]
```diff
@@ -10,23 +10,23 @@ Gem::Specification.new do |s|
...
-  s.add_dependency "workarea-swatches", "~> 1.x"
-  s.add_dependency 'workarea-search_autocomplete', '~> 1.x'
+  s.add_dependency 'workarea', '>= 3.5.0'
+  s.add_dependency 'workarea-theme', '>= 1.1.1'
+
+  s.add_dependency 'workarea-product_quickview', '>= 2.0.2'
+  s.add_dependency 'workarea-slick_slider', '>= 1.0.3'
+  s.add_dependency 'workarea-blog', '>= 3.5.0'
+  s.add_dependency 'workarea-content_search', '>= 1.1.0'
+  #s.add_dependency 'workarea-filter_dropdowns', '>= 3.0.2'
```
Comment (meowsus) [inline workarea-one_theme.gemspec]: Should we OSS this?

Comment (bencrouse) [reply workarea-one_theme.gemspec]: Yes

**Review by bencrouse — CHANGES_REQUESTED** [review]
Looks good, can you add the new search autocomplete? This should be a default included in all themes.

File: workarea-one_theme.gemspec (hunk 1 above)
Comment (meowsus) [reply workarea-one_theme.gemspec]: I done did it

File: workarea-one_theme.gemspec [hunk 2]
```diff
@@ -7,26 +7,27 @@ require "workarea/one_theme/version"
...
-  s.add_dependency "workarea-content_search", "~> 1.x"
-  s.add_dependency "workarea-filter_dropdowns", "~> 3.x"
-  s.add_dependency "workarea-gift_cards", "~> 4.x"
-  s.add_dependency "workarea-share", "~> 1.x", ">= 1.2.x"
-  s.add_dependency "workarea-reviews", "~> 3.x"
-  s.add_dependency "workarea-super_hero", "~> 2.x", ">= 2.1.x"
-  s.add_dependency "workarea-wish_lists", "~> 3.x"
-  s.add_dependency "workarea-swatches", "~> 1.x"
-  s.add_dependency 'workarea-search_autocomplete', '~> 1.x'
+  s.add_dependency 'workarea', '>= 3.5.0'
```
Comment (bencrouse) [inline workarea-one_theme.gemspec:18]: Any reason to not allow prereleases here anymore?

**Review by bencrouse — CHANGES_REQUESTED** [review]
Looks like more bundler problems :(

File: test/models/workarea/segment/rules/traffic_referrer_test.decorator [hunk 3]
```diff
@@ -0,0 +1,10 @@
+require 'test_helper'
+
+module Workarea
+  decorate Segment::Rules::TrafficReferrerTest, with: :one_theme do
```
Comment (bencrouse) [inline test/models/workarea/segment/rules/traffic_referrer_test.decorator]: I don't see how the theme could be affecting this but seems like we should fix in base?

**Review by bencrouse — CHANGES_REQUESTED** [review]
What's up with the bundle problems in the build?

File: workarea-one_theme.gemspec (hunk 2 above)
Comment (meowsus) [reply workarea-one_theme.gemspec:18]: Like I said in the NVY PR, I want to see if I can solve the prerelease problem with the Upgrade plugin using `bundle outdated --pre`. When we push up a beta for a new Workarea version it won't be compatible with the theme OOTB, so locking it down to stable versions feels safer to me. If it turns out that this problem can't be solved with the Upgrade plugin then I'll go back in these themes and enable pre-releases in each.

File: test/models/workarea/segment/rules/traffic_referrer_test.decorator (hunk 3 above)
Comment (meowsus) [reply test/models/workarea/segment/rules/traffic_referrer_test.decorator]: Yeah, I hit this at the end of the day when everyone was home already. I wanted to see if anyone else hit this issue before. I didn't know what else to do other than skip here. Let me know how you'd like me to resolve.

**Comment by meowsus** [comment]
> What's up with the bundle problems in the build?

First we were seeing this problem with `workarea-filter_dropdowns` just after I released it on RubyGems. Locally, I was able to bundle install/update just fine after I pushed the current version of the plugin to the gem server, but in the CI would fail saying that it couldn't find the plugin in any of the sources. We talked about it for a while and, when I kicked off the build again, the problem mystically resolved itself.

But it failed, again, because of a dependency `workarea-filter_dropdowns` required, namely `workarea-accordions`. I pushed that gem to the gem server too, yesterday morning, but the CI still hasn't figured out where it is. I know where it is. [It's right where it should be](https://rubygems.org/gems/workarea-accordions).

The CI error is telling us to use `--full-index` with our `bundle install` command. TIL `--full-index` does the following:

> This downloads the index all at once instead of making numerous small requests to the api.

This is touted as a solution for when "Bundler is really, really slow," according to their FAQ. Maybe this is a good idea in general, since we're trying to scrimp on our build minutes anyway.

@bencrouse would you like me to update workarea-commerce/ci to see if it fixes my issue and puts a few extra nanopennies in our proverbial pockets?

File: test/models/workarea/segment/rules/traffic_referrer_test.decorator (hunk 3 above)
Comment (bencrouse) [reply test/models/workarea/segment/rules/traffic_referrer_test.decorator]: Ok, let's do the fix in base - it should be written to handle `nil` situations anyways.

**Pull request merged**